desc/objectives/_geometry.py

```diff
@@ -750,24 +795,324 @@ def compute(self, equil_params, surface_params=None, constants=None):
         else:
-            surface_coords = compute_fun(
+            data_surf = compute_fun(
                 self._surface,
                 self._surface_data_keys,
                 params=surface_params,
                 transforms=constants["surface_transforms"],
                 profiles={},
                 basis="xyz",
-            )["x"]
-        d = safenorm(plasma_coords[:, None, :] - surface_coords[None, :, :], axis=-1)
+            )
+
+        surface_coords = data_surf["x"]
+        diff_vec = plasma_coords[:, None, :] - surface_coords[None, :, :]
+        d = safenorm(diff_vec, axis=-1)
+
+        point_signs = jnp.ones(surface_coords.shape[0])
+        if self._use_signed_distance:
+            surface_coords_rpz = xyz2rpz(surface_coords)
+
+            def _find_angle_vec(R, Z, Rtest, Ztest):
+                # R Z and surface points,
+                # Rtest Ztest are the point we wanna check is inside
+                # the surface or not
+                Rbool = R[:, None] > Rtest
+                Zbool = Z[:, None] > Ztest
+                # these Rbool are now size (Nsurf, Ntest)
+                quads = jnp.zeros_like(Rbool)
+                quads = jnp.where(jnp.logical_and(Rbool, Zbool), 0, quads)
+                quads = jnp.where(
+                    jnp.logical_and(jnp.logical_not(Rbool), Zbool), 1, quads
+                )
+                quads = jnp.where(
+                    jnp.logical_and(jnp.logical_not(Rbool), jnp.logical_not(Zbool)),
+                    2,
+                    quads,
+                )
+                quads = jnp.where(
+                    jnp.logical_and(Rbool, jnp.logical_not(Zbool)), 3, quads
+                )
+                deltas = quads[1:, :] - quads[0:-1, :]
+                deltas = jnp.where(deltas == 3, -1, deltas)
+                deltas = jnp.where(deltas == -3, 1, deltas)
+                # then flip sign if the R intercept is > Rtest and the
+                # quadrant flipped over a diagonal
+                b = (Z[1:] / R[1:] - Z[0:-1] / R[0:-1]) / (Z[1:] - Z[0:-1])
+                Rint = Rtest[:, None] - b * (R[1:] - R[0:-1]) / (Z[1:] - Z[0:-1])
+                deltas = jnp.where(
+                    jnp.logical_and(jnp.abs(deltas) == 2, Rint > Rtest),
+                    -deltas,
+                    deltas,
+                )
+                pt_sign = jnp.sum(deltas, axis=0)
+                # positive distance if the plasma pt is inside the surface, else
+                # negative distance is assigned
+                # pt_sign = 0 : Means SURFACE is OUTSIDE of the PLASMA,
+                #               assign positive distance
+                # pt_sign = +/-4: Means SURFACE is INSIDE PLASMA, so
+                #                 assign negative distance
+                pt_sign = jnp.where(jnp.isclose(pt_sign, 0), 1, -1)
+                return pt_sign
+
+            # loop over zeta planes
+            for plasma_zeta_idx, surface_zeta_idx in zip(
```

Comment:
can this be done with vmap or a jax loop?

Reply:
likely yes, I will refactor